patches/099-sqlite_api_3_26_0.patch

```diff
@@ -133,21 +462,650 @@ index d7477f57dbfca926fb8c8c78c63cb02a810e5c05..42b051da326d64d2f9fc920975a28316
+-      db_path().DirName().Append(FILE_PATH_LITERAL("sqlite-diag")));
++  base::FilePath breadcrumb_path =
++      db_path().DirName().AppendASCII("sqlite-diag");
+ 
+   // No stale diagnostic store.
+   ASSERT_TRUE(!base::PathExists(breadcrumb_path));
+diff --git a/sql/internal_api_token.h b/sql/internal_api_token.h
+new file mode 100644
+index 000000000000..bb92747426f8
+--- /dev/null
++++ b/sql/internal_api_token.h
+@@ -0,0 +1,25 @@
++// Copyright 2018 The Chromium Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style license that can be
++// found in the LICENSE file.
++
++#ifndef SQL_INTERNAL_API_TOKEN_H_
++#define SQL_INTERNAL_API_TOKEN_H_
++
++namespace sql {
++
++// Restricts access to APIs internal to the //sql package.
++//
++// This implements Java's package-private via the passkey idiom.
++class InternalApiToken {
++ private:
++  // Must NOT be =default to disallow creation by uniform initialization.
++  InternalApiToken() {}
++  InternalApiToken(const InternalApiToken&) = default;
++
++  friend class ConnectionTestPeer;
++  friend class Recovery;
++};
++
++}  // namespace sql
++
++#endif  // SQL_INTERNAL_API_TOKEN_H_
 diff --git a/sql/recovery.cc b/sql/recovery.cc
-index 64eefa51104efcced21f8caa0ab03a7da92d256a..0fac334df866ac10b8130764602b2ef193b37d3d 100644
+index 64eefa51104e..755d9fb77f9c 100644
 --- a/sql/recovery.cc
 +++ b/sql/recovery.cc
+@@ -154,19 +154,20 @@ std::unique_ptr<Recovery> Recovery::Begin(Connection* connection,
+   // same recovery.
+   if (!connection->is_open()) {
+     // Warn about API mis-use.
+-    DLOG_IF(FATAL, !connection->poisoned_)
++    DCHECK(connection->poisoned(InternalApiToken()))
+         << "Illegal to recover with closed database";
+     return std::unique_ptr<Recovery>();
+   }
+ 
+-  std::unique_ptr<Recovery> r(new Recovery(connection));
+-  if (!r->Init(db_path)) {
++  // Using `new` to access a non-public constructor
++  std::unique_ptr<Recovery> recovery(new Recovery(connection));
++  if (!recovery->Init(db_path)) {
+     // TODO(shess): Should Init() failure result in Raze()?
+-    r->Shutdown(POISON);
++    recovery->Shutdown(POISON);
+     return std::unique_ptr<Recovery>();
+   }
+ 
+-  return r;
++  return recovery;
+ }
+ 
+ // static
+@@ -191,8 +192,7 @@ Recovery::Recovery(Connection* connection)
+     : db_(connection),
+       recover_db_() {
+   // Result should keep the page size specified earlier.
+-  if (db_->page_size_)
+-    recover_db_.set_page_size(db_->page_size_);
++  recover_db_.set_page_size(db_->page_size());
+ 
+   // Files with I/O errors cannot be safely memory-mapped.
+   recover_db_.set_mmap_disabled();
 @@ -244,7 +244,7 @@ bool Recovery::Init(const base::FilePath& db_path) {
    }
  
    // Enable the recover virtual table for this connection.
 -  int rc = recoverVtableInit(recover_db_.db_);
-+  int rc = chrome_sqlite3_recoverVtableInit(recover_db_.db_);
++  int rc = recoverVtableInit(recover_db_.db(InternalApiToken()));
```

Comment:
Should be `chrome_sqlite3_recoverVtableInit(recover_db_.db(InternalApiToken()))`